### PR TITLE
Resolve issue setting updatedBy/At on update

### PR DESCRIPTION
**Pull request by kyle1morel**
- Adds a new global DB function **set_updatedAt()**. This simply sets the updatedAt field to the current time.
- Triggers added for each table calling this function on **before update**
- Passing correct user ID to Prisma update functions to set the updatedBy field

# Description

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/src/db/migrations/20231212000000_init.ts [hunk 1]
```diff
@@ -142,6 +180,16 @@ export async function up(knex: Knex): Promise<void> {
           table.unique(['permitId', 'permitTypeId', 'submissionId']);
         })
       )
+      .then(() =>
+        knex.schema.raw(`create trigger before_insert_permit_trigger
+          before insert on public.permit
```
Comment (TimCsaky) [inline app/src/db/migrations/20231212000000_init.ts:185]: just checking, did you want to set updatedAt for inserts to this table, or is it just updates?

File: app/src/db/migrations/20231212000000_init.ts [hunk 2]
```diff
@@ -510,12 +558,24 @@ export async function down(knex: Knex): Promise<void> {
...
       .then(() => knex.schema.dropTableIfExists('submission'))
+      .then(() => knex.schema.raw('DROP TRIGGER IF EXISTS before_update_user_trigger ON "user"'))
       .then(() => knex.schema.dropTableIfExists('user'))
+      .then(() =>
+        knex.schema.raw('DROP TRIGGER IF EXISTS before_update_identity_provider_trigger ON identity_provider')
+      )
       .then(() => knex.schema.dropTableIfExists('identity_provider'))
+      // Drop public schema PCNS functions
+      .then(() => knex.schema.raw('DROP FUNCTION IF EXISTS public.set_updatedAt'))
+      .then(() => knex.schema.raw('DROP FUNCTION IF EXISTS public.set_createdAt'))
```
Comment (TimCsaky) [inline app/src/db/migrations/20231212000000_init.ts:579]: i dont see set_createdAt function anywhere.

File: app/src/components/utils.ts [hunk 3]
```diff
@@ -68,6 +70,39 @@ export function getGitRevision(): string {
...
+/**
+ * @function getCurrentIdentity
+ * Attempts to acquire current identity value.
+ * Always takes first non-default value available. Yields `defaultValue` otherwise.
+ * @param {object} currentUser The express request currentUser object
+ * @param {string} [defaultValue=undefined] An optional default return value
+ * @returns {string} The current user identifier if applicable, or `defaultValue`
+ */
+export function getCurrentIdentity(currentUser: CurrentUser | undefined, defaultValue: string | undefined = undefined) {
+  return parseIdentityKeyClaims()
```
Comment (TimCsaky) [inline app/src/components/utils.ts:82]: i always thought this was overly complicated in COMS and i didnt like all the NIL's and defaultValue stuff.
could be one function 
```js
getCurrentUserId(currentUser){
// get claim name matching config.identityKey
// get claim value from currentUser.tokenPayload
// do database query to return userId
}
```
that being said.. if this works maybe it's better for now.

File: app/src/db/migrations/20231212000000_init.ts (hunk 1 above)
Comment (kyle1morel) [reply app/src/db/migrations/20231212000000_init.ts:185]: Yeah, needs to be set on insert for sorting purposes.

File: app/src/components/utils.ts (hunk 3 above)
Comment (kyle1morel) [reply app/src/components/utils.ts:82]: I agree. Might try to clean it up during the refactoring process before going to prod.

File: app/src/db/migrations/20231212000000_init.ts (hunk 2 above)
Comment (kyle1morel) [reply app/src/db/migrations/20231212000000_init.ts:579]: Good catch. Was an old function I removed.

**Review by wilwong89 — APPROVED** [review]

**Review by TimCsaky — APPROVED** [review]

**Pull request merged**